### ISSUE TITLE
google-closure-compiler-osx crashes on Ventura

### ISSUE THREAD

**Issue by gonfunko** [google/closure-compiler-npm#265]
`npm run build:compiled` in [Blockly](https://github.com/google/blockly), which depends on Closure, is failing on a macOS Ventura machine with the following output:

```
[15:02:04] Using gulpfile ~/Developer/blockly/gulpfile.js
[15:02:04] Starting 'buildCompiled'...
[15:02:06] gulp-google-closure-compiler: Error writing to stdin of the compiler. write EPIPE
[15:02:06] 'buildCompiled' errored after 2.09 s
[15:02:06] Error in plugin 'gulp-google-closure-compiler'
Message:
    Compilation errors occurred
Details:
    domainEmitter: [object Object]
    domainThrown: false
```

This causes a crashlog (attached) for the google-closure-compiler-osx binary to be generated, and running `./compiler` directly fails with `fish: Job 1, './compiler' terminated by signal SIGSEGV (Address boundary error)` and an accompanying crashlog.
[compiler-2022-10-28-150211.txt](https://github.com/google/closure-compiler-npm/files/9892199/compiler-2022-10-28-150211.txt)

**Comment by ChadKillingsworth** [comment]
What version are you using? I've been using MacOS Ventura for about 3 months without issue.

**Comment by ChadKillingsworth** [comment]
Also - what architecture Mac are you experiencing this on? (are you on an M1)

**Comment by grantila** [comment]
Can confirm this. I'm on a 2018 i7 with Ventura.

```
$ node_modules/google-closure-compiler-osx/compiler

[1]    43748 segmentation fault  node_modules/google-closure-compiler-osx/compiler
```

**Comment by ChadKillingsworth** [comment]
I'm not able to recreate this issue. The downloaded binaries are executing just fine for me (using an M1). I am on Ventura 13.1 Beta, but it seems odd that would make a difference.

You might try looking at different versions and seeing if any of the older release versions work. It would be interesting to know what version this started occurring with.

**Comment by ChadKillingsworth** [comment]
You might also try the Mac OS version found at the bottom of this build: https://github.com/google/closure-compiler-npm/actions/runs/3427942157

**Comment by grantila** [comment]
Same with an older version 20220502.0.0. The version at the bottom of that page segfaults too.

I wonder why this is, could it be related to XCode or homebrew libs? I'm on XCode 14.1 (14B47b). The compiler is stat-linked, so it's quite surprising...

**Comment by ray007** [comment]
Same here, tried with version `20220601.0.0`, `20221004.0.0` and `20221102.0.0`.

Machine is a Mac mini with Intel Core i7 CPU.

**Comment by ChadKillingsworth** [comment]
I've been running on Ventura betas without issue since mid-July. However I do have other packages installed via homebrew so its possible this is indeed a library issue. I haven't seen any similar issues on the Graal github repo to help.

Based off your testing, it seems that the Graal version itself is not the issue.

**Comment by grantila** [comment]
Same, segmentation fault...

**Comment by ChadKillingsworth** [comment]
I'm wondering if this is upx https://github.com/upx/upx/issues/612

**Comment by grantila** [comment]
we could give it a shot!

**Comment by ChadKillingsworth** [comment]
Alright here's the binary without UPX compression: https://github.com/google/closure-compiler-npm/actions/runs/3488438003

**Comment by gonfunko** [comment]
I can confirm it's working now as well, thanks for tracking this down!

**Comment by ray007** [comment]
Can we please get a new release with this?
Thank you.

**Comment by ChadKillingsworth** [comment]
Fixed by #274

I created a patch release for this.
